### PR TITLE
Adding review test case has been created.

### PR DESCRIPTION
**Pull request by bambula4000**
Added review.py module in oct/pages for implementing objects on the product
review page.
Added review.py module in oct/tests/web for implementing and running test case
which verifies ability to add review on the product page.

### CODE REVIEW

File: oct/pages/review.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+from selenium.webdriver import Remote
+from oct.pages.base import Page
+
+
+class ProductPage(Page):
+    def __init__(self, browser: Remote) -> None:
```
Comment (AlekseyDidilov) [inline oct/pages/review.py]: Pay attention to the first comment in PR #6.

File: oct/pages/review.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        self._browser.get(
+            "http://localhost/index.php?route=product/product&path=25_28&product_id=33"
+        )
+
+    def loaded(self) -> bool:
+        return "Samsung SyncMaster 941BW" in self._browser.title
+
+
+class AddReviewPage:
+    def __init__(self, browser: Remote):
```
Comment (AlekseyDidilov) [inline oct/pages/review.py]: Methods within the AddReviewPage are methods ProductPage, isn't it?

File: oct/tests/web/review.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        chrome: Remote = Chrome(grid)
+        product_page = ProductPage(chrome)
+        product_page.open()
+        review_tab = AddReviewPage(chrome)
+        review_tab.review_link()
+        review_tab.type_name("AutoTestBot")
+        review_tab.type_review("This review has been wrote by auto test bot!")
+        review_tab.rating_value()
+        review_tab.press_continue()
+
```
Comment (AlekseyDidilov) [inline oct/tests/web/review.py]: You don't check anything here.

File: oct/pages/review.py (hunk 1 above)
Comment (bambula4000) [reply oct/pages/review.py]: Thanks. It's done.

File: oct/pages/review.py (hunk 2 above)
Comment (bambula4000) [reply oct/pages/review.py]: Thanks. It's done.

File: oct/tests/web/review.py (hunk 3 above)
Comment (bambula4000) [reply oct/tests/web/review.py]: Thanks. 
It's done.

File: oct/tests/web/review.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+class UsersReview(Testcase):
+    @test
+    def test(self, grid: str) -> None:
+        chrome: Remote = Chrome(grid)
+        product_page = ProductPage(chrome, "33", "Samsung SyncMaster 941BW")
+        product_page.open()
+        product_page.review_link()
+        review_tab = FillReview(chrome)
```
[the remark this answers is not in the document]
Comment (bambula4000) [reply oct/tests/web/review.py]: Thanks. Has been done.

File: oct/pages/product_page.py [hunk 5]
```diff
@@ -23,3 +23,9 @@ def open_brand_page(self) -> None:
             '//*[@id="content"]/div[1]/div[2]/ul[1]/li[1]/a'
         )
         brand_name.click()
+
+    def review_link(self) -> None:
+        review_tab = self._browser.find_element_by_xpath(
```
[the remark this answers is not in the document]
Comment (bambula4000) [reply oct/pages/product_page.py]: Thanks. Has been done.

File: oct/pages/review.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        name_field.click()
+        name_field.send_keys(name)
+
+    def type_review(self, review: str) -> None:
+        review_field = self._browser.find_element_by_id("input-review")
+        review_field.click()
+        review_field.send_keys(review)
+
+    def rating_value(self) -> None:
+        rating_value = self._browser.find_element_by_xpath(
```
[the remark this answers is not in the document]
Comment (bambula4000) [reply oct/pages/review.py]: Thanks. Has been done.

File: oct/tests/web/review.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        review_tab.type_name("AutoTestBot")
+        review_tab.type_review("This review has been wrote by auto test bot!")
+        review_tab.rating_value()
```
[the remark this answers is not in the document]
Comment (bambula4000) [reply oct/tests/web/review.py]: Thanks. Has been done.

File: oct/pages/review.py [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+            '//*[@id="form-review"]/div[4]/div/input[5]'
+        )
+        rating_value.click()
+
+    def press_continue(self) -> None:
+        self._browser.find_element_by_class_name("btn-primary").click()
+
+    def success_adding(self) -> bool:
+        return self._browser.find_element_by_class_name("alert-success")
```
[the remark this answers is not in the document]
Comment (bambula4000) [reply oct/pages/review.py]: Thanks. Has been done.

File: oct/tests/web/review.py [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+from oct.pages.product_page import ProductPage
+
+
+class UsersReview(Testcase):
+    @test
+    def test(self, grid: str) -> None:
+        chrome: Remote = Chrome(grid)
+        product_page = ProductPage(chrome, "33", "Samsung SyncMaster 941BW")
+        product_page.open()
+        product_page.review_link()
```
Comment (extsoft) [inline oct/tests/web/review.py]: @bambula4000 `review_link` has to return something. Please rename.

File: oct/tests/web/review.py [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+        review_tab = Review(chrome)
+        review_tab.type_name("AutoTestBot")
+        review_tab.type_review("This review has been wrote by auto test bot!")
+        review_tab.choose_rating()
```
Comment (extsoft) [inline oct/tests/web/review.py]: @bambula4000 I expect to see which rating you re going to select.

File: oct/tests/web/review.py (hunk 9 above)
Comment (bambula4000) [reply oct/tests/web/review.py]: what do you mean, i dont understand?

Comment (extsoft) [reply oct/tests/web/review.py]: If a name of a method is a noun (`product_page.review_link`), the method should return something. If a method has no return, a name of this method has to be a verb.

Comment (bambula4000) [reply oct/tests/web/review.py]: oh, now i'm understand. Thank.

File: oct/tests/web/review.py (hunk 10 above)
Comment (bambula4000) [reply oct/tests/web/review.py]: its done

File: oct/pages/review.py [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+            f'//*[@id="form-review"]/div[4]/div/input' f"[{rating}]"
+        )
+        rating_s.click()
+        rating_s.send_keys(rating)
+
+    def press_continue(self) -> None:
+        self._browser.find_element_by_class_name("btn-primary").click()
+
+    def success_adding(self) -> bool:
+        if self._browser.find_element_by_class_name("alert-success"):
```
Comment (extsoft) [inline oct/pages/review.py]: @bambula4000 An exception will be thrown if the element is not found. So, this method can return True or thrown an error. But I think if you can't find an element, you have to return False.

File: oct/pages/review.py [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+        rating_s = self._browser.find_element_by_xpath(
+            f'//*[@id="form-review"]/div[4]/div/input' f"[{rating}]"
+        )
+        rating_s.click()
+        rating_s.send_keys(rating)
+
+    def press_continue(self) -> None:
+        self._browser.find_element_by_class_name("btn-primary").click()
+
+    def success_adding(self) -> bool:
```
Comment (extsoft) [inline oct/pages/review.py]: @bambula4000 Please rename to `successfully_added`.

File: oct/pages/review.py [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+        )
+        rating_s.click()
+        rating_s.send_keys(rating)
+
+    def press_continue(self) -> None:
+        self._browser.find_element_by_class_name("btn-primary").click()
+
+    def successfully_added(self) -> bool:
+        try:
+            return self._browser.find_element_by_class_name("alert-success")
```
Comment (extsoft) [inline oct/pages/review.py]: @bambula4000 you return a `WebElement` instance, but I expect `bool` object.

**Comment by extsoft** [comment]
@bambula4000 
```bash
git fetch
git rebase origin/master
# fix conflicts
git push -f
``` is what you have to do

**Pull request merged**